### PR TITLE
update docs for xgboost1.7.1 and add python notebooks

### LINKED ISSUE #241: since dmlc xgboost 1.7.0 released, we need to update the codes/docs/notebooks
since dmlc xgboost 1.7.0 released, we need to:
1. update the codes
2. add the python notebooks
3. update the docs

### LINKED ISSUE #244: AWS getting started doc is out of date
we need to update AWS getting started doc to request users to download Fannie Mae dataset.

### LINKED ISSUE #250: we need to update doc if spark-rapids support arm CPU architecture
since we support CPU arm arch from v22.12, we need to update docs(maybe some scripts like [makefile](https://github.com/NVIDIA/spark-rapids-examples/blob/main/examples/UDF-Examples/Spark-cuSpatial/src/main/native/CMakeLists.txt)). see #247

### PR DESCRIPTION
**Pull request by nvliyuan**
Signed-off-by: liyuan <yuali@nvidia.com>
this pr is to:
1. update getting started docs for xgboost1.7.0
2. add python notebooks
3. update out of date aws doc fix #244 
4. update xgboost1.7.0 performance fix #241

### CODE REVIEW

File: docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md [hunk 1]
```diff
@@ -47,6 +49,13 @@ And here are the steps to enable the GPU resources discovery for Spark 3.1+.
     spark.worker.resource.gpu.amount 1
     spark.worker.resource.gpu.discoveryScript ${SPARK_HOME}/examples/src/main/scripts/getGpusResources.sh
     ```
+3. Install the XGBoost, cudf-cu11, numpy libraries on all nodes before running XGBoost application.
+
+``` bash
+pip install xgboost==1.7.0
```
Comment (wbo4958) [inline docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md:55]: pip install xgboost

Comment (nvliyuan) [reply docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md]: updated

File: examples/XGBoost-Examples/agaricus/notebooks/python/agaricus-gpu.ipynb [hunk 2]
```diff
@@ -88,9 +146,18 @@
     "schema = StructType([ StructField(x, FloatType()) for x in [label] + features ])\n",
     "\n",
     "# You need to update them to your real paths!\n",
-    "dataRoot = os.getenv(\"DATA_ROOT\", \"/data\")\n",
-    "train_data = reader.schema(schema).option('header', True).csv(dataRoot + '/agaricus/csv/train')\n",
-    "trans_data = reader.schema(schema).option('header', True).csv(dataRoot + '/agaricus/csv/test')"
+    "dataRoot = os.getenv(\"DATA_ROOT\", \"/your-path\")\n",
+    "train_path = dataRoot + \"/your-path\"\n",
```
Comment (NvTimLiu) [inline examples/XGBoost-Examples/agaricus/notebooks/python/agaricus-gpu.ipynb:150]: This part is a little confused, as `train_path=eval_path=/your-path/your-path` by default.

Based on datasets provided : https://github.com/NVIDIA/spark-rapids-examples/blob/main/datasets/agaricus-small.tar.gz, suggest to change as below, which also align with Scala test code 

`dataRoot = os.getenv("DATA_ROOT", "/agaricus/csv/train")`

https://github.com/NVIDIA/spark-rapids-examples/blob/main/examples/XGBoost-Examples/agaricus/notebooks/scala/agaricus-gpu.ipynb
```

// You need to update them to your real paths!
val dataRoot = sys.env.getOrElse("DATA_ROOT", "/data")
val trainPath = dataRoot + "/agaricus/csv/train/"
val evalPath  = dataRoot + "/agaricus/csv/test/"
val transPath = dataRoot + "/agaricus/csv/test/"
```

Or, just overwrite them with ENVs like

```
train_path = os.getenv("TRAIN_DATASET_PATH", 'xxx_train')
eval_path = os.getenv("EVAL_DATASET_PATH", 'xxx_eval')
```

Comment (nvliyuan) [reply examples/XGBoost-Examples/agaricus/notebooks/python/agaricus-gpu.ipynb:150]: I tested in local/internal clusters and I did not consider the ease-to-test for CICD jobs, as discussed with @NvTimLiu,  feel free to update the data paths/number workers.

**Review by NvTimLiu — APPROVED** [review]
CI job against the notebooks got PASS, +1

File: docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md [hunk 3]
```diff
@@ -47,6 +49,13 @@ And here are the steps to enable the GPU resources discovery for Spark 3.1+.
     spark.worker.resource.gpu.amount 1
     spark.worker.resource.gpu.discoveryScript ${SPARK_HOME}/examples/src/main/scripts/getGpusResources.sh
     ```
+3. Install the XGBoost, cudf-cu11, numpy libraries on all nodes before running XGBoost application.
+
+``` bash
+pip install xgboost
```
Comment (wbo4958) [inline docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md:55]: Do we need to install scikit-learn?

File: docs/get-started/xgboost-examples/on-prem-cluster/yarn-python.md [hunk 4]
```diff
@@ -32,6 +34,30 @@ We use `SPARK_HOME` environment variable to point to the Apache Spark cluster.
 And as to how to enable GPU scheduling and isolation for Yarn,
 please refer to [here](https://hadoop.apache.org/docs/r3.1.0/hadoop-yarn/hadoop-yarn-site/UsingGpus.html).
 
+Please make sure to install the XGBoost, cudf-cu11, numpy libraries on all nodes before running XGBoost application.
+``` bash
+pip install xgboost
+pip install cudf-cu11 --extra-index-url=https://pypi.ngc.nvidia.com
+pip install numpy
```
Comment (wbo4958) [inline docs/get-started/xgboost-examples/on-prem-cluster/yarn-python.md:41]: same with the previous comment

File: examples/XGBoost-Examples/README.md [hunk 5]
```diff
@@ -1,19 +1,18 @@
...
+Spark XGBoost examples here showcase the need for ETL+Training pipeline GPU acceleration.
 The Scala based XGBoost examples here use [DMLC’s version](https://repo1.maven.org/maven2/ml/dmlc/xgboost4j-spark_2.12/).
-For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that
-uses [NVIDIA’s Spark XGBoost version](https://repo1.maven.org/maven2/com/nvidia/xgboost4j-spark_3.0/1.4.2-0.3.0/).
+The pyspark based XGBoost examples requires [installing RAPIDS via pip](https://rapids.ai/pip.html#install).
 Most data scientists spend a lot of time not only on
 Training models but also processing the large amounts of data needed to train these models.
-As you can see below, XGBoost training on GPUs can be up to 10X and data processing using
-RAPIDS Accelerator can also be accelerated with an end-to-end speed-up of 7X on GPU compared to CPU.
+As you can see below, Pyspark+XGBoost training on GPUs can be up to 13X and data processing using
```
Comment (wbo4958) [inline examples/XGBoost-Examples/README.md:8]: Can we also have a benchmark testing for xgboost-jvm-gpu?

**Review by wbo4958 — COMMENTED** [review]
Overall, it LGTM.

File: docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md (hunk 3 above)
Comment (nvliyuan) [reply docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md:55]: yes, add 'pip install scikit-learn '

File: docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md [hunk 6]
```diff
@@ -47,6 +49,13 @@ And here are the steps to enable the GPU resources discovery for Spark 3.1+.
...
+pip install cudf-cu11 --extra-index-url=https://pypi.ngc.nvidia.com
+pip install numpy
```
[the remark this answers is not in the document]
Comment (nvliyuan) [reply docs/get-started/xgboost-examples/on-prem-cluster/standalone-python.md:57]: yes, XGBoost dependent on numpy
image

File: examples/XGBoost-Examples/README.md (hunk 5 above)
Comment (nvliyuan) [reply examples/XGBoost-Examples/README.md:8]: no, but I think we can add it in another PR.

**Review by wbo4958 — APPROVED** [review]

**Pull request merged**